### DISCUSSION TITLE
Several Icons on the same IconLayer

### DISCUSSION THREAD

**Discussion by ventoline** [visgl/deck.gl#8163] [Q&A] [answered]
Is it possible to display several different Icons in the same layer?

` iconAtlas: d => ( icons[treeFamilies.indexOf(d.properties.family)] ) 
`  
doesnt work, or any randomly assigned texture

**Comment by Pessimistress** [comment]
Read [the documentation](https://deck.gl/docs/api-reference/layers/icon-layer)

**Comment by ventoline** [comment]
I did, many times. there isn't anything on using an array of icons, and the examples don't provide it. My answer isn't in there unfortunately. Also a note on the documentation, the examples dont always compile and in some cases it feels incomplete.

**Reply by Pessimistress** [reply]
Well I suggest you read the properties section. `iconAtlas` is a single image (if you choose the prepacked route). `iconMapping` defines where multiple icons can be located in that image. You specify which icons to render using `getIcon`.

**Reply by ventoline** [reply, marked as answer]
thank you for your reply. I dont have a say in the route,i am given separate image files andneed attach them to the Icon.
 This seems to work: 
```
  new IconLayer({
        id: "trees",
      data: data2.features, 
      getIcon: d =>({ url:  icons[treeFam2Valid.indexOf(d.properties.family)],
     width: 128,
     height: 128,
     mask:true //needed for colors !!!
    }), 
      getPosition:  d => d.geometry.coordinates ,
      getSize: d => ((d.properties.CrownRad + 1) * 3 ),
      sizeScale: 1,
      sizeUnits : "meters",
      sizeMinPixels : 12,
      getAngle : d => (Math.random()*360 ),   
     
       getColor: d => (d.geometry.coordinates[1] < 43.6293  &&  d.geometry.coordinates[0] < -79.418)?  /* && d.properties.SP_CODE !== "DEAD" */ 
       [
        colors[treeTypes.indexOf(d.properties.tag)][0],
        colors[treeTypes.indexOf(d.properties.tag)][1], 
        colors[treeTypes.indexOf(d.properties.tag)][2],
        (selectedOption === "all") || (selectedOption === d.properties.family) ?  220: 20
       ] 
         : [0,0,0,0] 
       ,
         updateTriggers: {
          getColor: [selectedOption]
         },  
      })
 
```
